mbed_build/_internal/config/source.py

```diff
@@ -21,17 +22,15 @@ class Source:
     - mbed_app.json
 
     This class provides a common interface to configuration data.
-    It also handles data normalisation, to fix inconsistencies between sources:
-    - namespacing
-    - override filtering for specific target labels
     """
 
-    name: str
+    human_name: str
     config: dict
-    target_overrides: dict
+    config_overrides: dict
+    cumulative_overrides: dict
 
     @classmethod
-    def from_mbed_lib(cls, file: Path, target_labels: List[str]) -> "Source":
+    def from_mbed_lib(cls, file: Path, target_labels: Iterable[str]) -> "Source":
```

Comment:
Are target labels here all possible target attributes that can be overridden, or just the cumulative ones? I have lost the idea of what is an `allowed label` in `_filter_target_overrides`.

Reply:
I thought target labels are only the ones that identify target type? 
I.e.: `STM`, `CORTEX_A`, etc.
Maybe we need better names. 🤔 

It's hard to differentiate between different label types.

Reply:
Ah, yes sorry so its anything returned by `target.labels`? That makes sense.

Reply:
This makes sense in this code I reckon. I was just confusing it with 'labels' in the old code which is used to both mean `targets.labels` and the 'label' used to create the name of a magic directory. Its better to keep it to the narrower meaning I reckon, like you are doing.

Reply:
That is right. 
Additionally they can be accumulated during configuration parsing. Old tools seem to allow it.

Reply:
Yeah 'labels' is one of the cumulative attributes.

Reply:
There's also `LabelFilter`, where for directories `label` has a slightly different meaning...

Reply:
Hmm so we do use for both meanings 🤔 . Its not ideal, but also not a biggie for me. Its mostly stemming from the name that we are giving the attribute in targets.json. ie. `target.labels` returns all the things that go into the 'LabelFilter' for `TARGET_<label>`.